src/platform/windows/events_loop.rs

```diff
@@ -306,6 +339,25 @@ pub unsafe extern "system" fn callback(window: winapi::HWND, msg: winapi::UINT,
                 window_id: SuperWindowId(WindowId(window)),
                 event: Resized(w, h),
             });
+
+            // Wait for the parent thread to process the resize event before returning from the
+            // callback.
+            CONTEXT_STASH.with(|context_stash| {
+                let mut context_stash = context_stash.borrow_mut();
+                let cstash = context_stash.as_mut().unwrap();
+
+                // If this window has been inserted into the window map, the resize event happened
+                // during the event loop. If it hasn't, the event happened on window creation and
+                // should be ignored.
+                if cstash.windows.get(&window).is_some() {
+                    let (ref mutex, ref cvar) = *cstash.win32_block_loop;
+                    let mut block_thread = mutex.lock().unwrap();
```

Comment:
You need to lock the mutex before calling `send_event`, otherwise it is possible that the event gets processed and `notify_all()` gets called before we even reach that point.

I think the best thing to do is to not call `send_event` here but directly `context_stash.as_ref().unwrap().sender.send(event)`.